src/main/java/com/auctionapp/api/model/dto/CategoryDto.java

```diff
@@ -0,0 +1,55 @@
+package com.auctionapp.api.model.dto;
+
+import java.util.Set;
+import java.util.UUID;
+
+public class CategoryDto {
+	private UUID id;
+	private String name;
+	private CategoryDto subcategory;
```

Comment:
Can't a category have several subcategories, not just one?

Reply:
Maybe I should rename that field as subcategory_of, this way it is misleading.
Because actually each subcategory will be created as a record of the category table and will have this field filled with the parent category id.